### PR TITLE
Enhancement/skale 4466 all interfaces

### PR DESCRIPTION
**Pull request by payvint**

### CODE REVIEW

File: contracts/Nodes.sol [hunk 1]
```diff
@@ -50,50 +52,12 @@ import "./Permissions.sol";
  * 
  * Note: Online nodes contain both Active and Leaving states.
  */
-contract Nodes is Permissions {
+contract Nodes is Permissions, INodes {
     
-    using Random for Random.RandomGenerator;
+    using Random for INodes.RandomGenerator;
```
Comment (DimaStebaev) [inline contracts/Nodes.sol]: Why is `RandomGenerator` declared in `INodes` interface?

File: contracts/utils/FieldOperations.sol [hunk 2]
```diff
@@ -24,19 +24,21 @@
...
 import "./Precompiled.sol";
 
 
 library Fp2Operations {
 
-    struct Fp2Point {
-        uint a;
-        uint b;
-    }
+    // struct ISkaleDKG.Fp2Point {
```
Comment (DimaStebaev) [inline contracts/utils/FieldOperations.sol]: This is not needed here.

File: contracts/utils/FieldOperations.sol [hunk 3]
```diff
@@ -153,22 +163,22 @@ library G1Operations {
...
 
 library G2Operations {
-    using Fp2Operations for Fp2Operations.Fp2Point;
+    using Fp2Operations for ISkaleDKG.Fp2Point;
 
-    struct G2Point {
-        Fp2Operations.Fp2Point x;
-        Fp2Operations.Fp2Point y;
-    }
+    // struct ISkaleDKG.G2Point {
```
Comment (DimaStebaev) [inline contracts/utils/FieldOperations.sol]: This is not needed here.

File: dictionaries/domain-terms.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 agpl
 ecdh
+IECDH
```
Comment (DimaStebaev) [inline dictionaries/domain-terms.txt]: Maybe `ECDH` but not `IECDH`?

Comment (DimaStebaev) [reply dictionaries/domain-terms.txt]: Cspell does not understand that `IECDH` is `I` + `ECDH`.

**Review by DimaStebaev — APPROVED** [review]

**Review by yavrsky — APPROVED** [review]

**Review by gannakulikova — APPROVED** [review]

**Pull request merged**